bpf/go_common.h

```diff
@@ -401,4 +401,37 @@ static __always_inline void *unwrap_tls_conn_info(void *conn_ptr, void *tls_stat
     return conn_ptr;
 }
 
+static __always_inline void process_meta_frame_headers(void *frame, tp_info_t *tp) {
+    if (frame) {
+        void *fields = 0;
+        bpf_probe_read(&fields, sizeof(fields), (void *)(frame + 8));
+        u64 fields_len = 0;
+        bpf_probe_read(&fields_len, sizeof(fields_len), (void *)(frame + 8 + 8));
+        bpf_dbg_printk("fields ptr %llx, len %d", fields, fields_len);
+        if (fields && fields_len > 0) {
+            for (int i = 0; i < 16; i++) {
+                if (i >= fields_len) {
```

Comment:
```suggestion
            for (u8 i = 0; i < 16; i++) {
                if (i >= fields_len) {
```
rationale: in this particular case, avoids signed/unsigned comparisons - overall I tend to favor unsigned sizes (unless `ssize_t` is warranted) as it plays more naturally with the rest of the language (*) and overflows are easier to deal with.

(*) see line 416 for an example, in which sizeof always returns `size_t`